### PR TITLE
clippyWrapper: Watch reset property to get reset button working.

### PR DESCRIPTION
**Pull request by cfoch**
https://phabricator.endlessm.com/T24930

### CODE REVIEW

**Comment by cfoch** [comment]
It works, but I am not sure if this is mergeable. I need feedback.

**Comment by cosimoc** [comment]
I'd like to understand why the roundtrip is needed.
Is it because otherwise you will receive no notification, and so `_onNotify()` here https://github.com/endlessm/hack-toolbox-app/blob/5818894715484e72e41e41fb01a6b78d14d0cef4/src/Fizzics/level2.js#L317 never gets called, or because `_onNotify()` somehow does not regenerate the code correctly? We should be able to invalidate the current codeview state entirely from within the toolbox, since that's what activates the reset action in the first place.

**Comment by cfoch** [comment]
Do not merge it yet.

**Comment by cfoch** [comment]
Yes, this is ready to merge.

File: src/Fizzics/toolbox.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ var FizzicsToolbox = GObject.registerClass(class FizzicsToolbox extends Toolbox
                 new GLib.Variant('(sava{sv})', ['reset', [], {}]),
                 null, Gio.DBusCallFlags.NONE, -1, null
             );
+            this._controlPanel._level2._onNotify();
```
Comment (cosimoc) [inline src/Fizzics/toolbox.js]: This is a bit punching a hole through the different layers...
The `FizzicsLevel2` class already has access to the model `reset` signal through the `bindModel()` method. I think that would be a much better place to connect to `reset` and call `_onNotify()` from.

**Review by cosimoc — CHANGES_REQUESTED** [review]

File: src/Fizzics/toolbox.js (hunk 1 above)
Comment (cfoch) [reply src/Fizzics/toolbox.js]: But the 'reset' signal is emitted by the Toolbox class. Not from the model.

Comment (cfoch) [reply src/Fizzics/toolbox.js]: And I have no access to the Toolbox from FizzicsLevel2.

**Comment by cosimoc** [comment]
One way to solve that could to add a reset method to the model and move there the code that is responsible for calling the action... or you could find another way to propagate the reset call down to the levels, either by giving them access to the toolbox or through calling a method directly.

**Pull request merged**